froide/bounce/utils.py

```diff
@@ -328,3 +333,127 @@ def handle_smtp_error(exc):
             continue
         raise exc
     return True
+
+
+def check_delivery_from_log():
+    """
+    check and update the delivery status of FOI-Messages
+
+    1. query model for all FOI-messages that have an update-able delivery state
+    2. collect new postfix-log entries
+    3. match msg-ids from postfix log with queries FOI-messages
+    4. update FOI-messages accordingly
+    """
+    POSTFIX_LOG_PATH = "/var/log/mail.log"
+    PYGTAIL_OFFSET_PATH = "./pt.offset"
+    FINAL_STATES = DeliveryStatus.FINAL_STATUS
+    RELEVANT_FIELDS = set(("message-id", "from", "to", "status"))
+
+    # query model
+    messages = (
+        FoiMessage.objects.exclude(status__in=FINAL_STATES)
```

Comment:
Watch out, `status` here is related to FoiRequest.status (which indicates the status of the whole request, status on message mirrors the status for that message on classification). What you're looking for is `deliverystatus__status`.